third_party/proxy/types.go

```diff
@@ -58,6 +58,8 @@ type Provider interface {
 	SyncLoop()
 	Run(stopCh <-chan struct{})
 	GetServiceByIP(serviceStr string) (ServicePortName, bool)
+	// IsInitialized tells whether both Endpoints and Services has been synced.
+	IsInitialized() bool
```

Comment:
I do not mean you must change, but just for your reference: in my PR I added a new interface to wrap and extend proxy.Provider to avoid modifying the copied code: https://github.com/vmware-tanzu/antrea/pull/1877/files#diff-ef59a1d8866e8f8a6bd438ff001db8eb4ddf6fe167a5f82a8389aa6a9969ca6bR49.

In my case, I added an more implementation specific func which probably makes little sense to add to the Provider interface, while IsInitialized() sounds much more generic. So, would let you decide. 
@weiqiangt

Reply:
I actually would like to follow your style. I plan to rebase this PR once #1877 merged.